src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/AddressServiceImpl.java

```diff
@@ -1,15 +1,95 @@
+import uk.gov.ons.ctp.integration.contactcentresvc.service.addressindex.model.AddressIndexAddressDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.service.addressindex.model.AddressIndexSearchResultsDTO;
 
 /**
  * A ContactCentreDataService implementation which encapsulates all business logic for getting
  * Addresses
  */
 @Service
+@Validated()
 public class AddressServiceImpl implements AddressService {
-  private static final Logger log = LoggerFactory.getLogger(AddressServiceImpl.class);
+
+  @Autowired private AddressServiceClientServiceImpl addressServiceClient;
+
+  @Override
+  public AddressQueryResponseDTO addressQuery(AddressQueryRequestDTO addressQueryRequest) {
+    // Delegate the query to Address Index
+    AddressIndexSearchResultsDTO addressIndexResponse =
+        addressServiceClient.addressQuery(addressQueryRequest);
+
+    // Summarise the returned addresses
+    return convertAddressIndexResultsToSummarisedAdresses(addressIndexResponse);
+  }
+
+  @Override
+  public AddressQueryResponseDTO postcodeQuery(PostcodeQueryRequestDTO postcodeQueryRequest) {
+    // Delegate the query to Address Index
+    AddressIndexSearchResultsDTO addressIndexResponse =
+        addressServiceClient.postcodeQuery(postcodeQueryRequest);
+
+    // Summarise the returned addresses
+    return convertAddressIndexResultsToSummarisedAdresses(addressIndexResponse);
+  }
+
+  private AddressQueryResponseDTO convertAddressIndexResultsToSummarisedAdresses(
+      AddressIndexSearchResultsDTO addressIndexResponse) {
+    ArrayList<AddressDTO> summarisedAddresses = new ArrayList<>();
+    for (AddressIndexAddressDTO fullAddress : addressIndexResponse.getResponse().getAddresses()) {
+      String formattedAddress = fullAddress.getFormattedAddress();
+      String addressPaf = fullAddress.getFormattedAddressPaf();
+      String addressNag = fullAddress.getFormattedAddressNag();
+      String welshAddressPaf = fullAddress.getWelshFormattedAddressPaf();
+      String welshAddressNag = fullAddress.getWelshFormattedAddressNag();
+
+      AddressDTO addressSummary = new AddressDTO();
+      addressSummary.setUprn(fullAddress.getUprn());
+      addressSummary.setFormattedAddress(
+          selectFirstUsableAddress(addressPaf, addressNag, formattedAddress));
+      addressSummary.setWelshFormattedAddress(
+          selectFirstUsableAddress(welshAddressPaf, welshAddressNag, formattedAddress));
+
+      summarisedAddresses.add(addressSummary);
+    }
+
+    // Complete construction of response objects
+    AddressQueryResponseDTO queryResponse = new AddressQueryResponseDTO();
+    queryResponse.setDataVersion(addressIndexResponse.getDataVersion());
+    queryResponse.setAddresses(summarisedAddresses);
+    queryResponse.setTotal(addressIndexResponse.getResponse().getTotal());
+
+    return queryResponse;
+  }
+
+  /**
+   * This method takes multiple addresses and returns the first usable one.
```

Comment:
Perfect candidate for a util message kept in service-common StringUtils
Is only selecting first usable address because it is only used here - could be selectFirstNonBlankString(...)

Reply:
Moved as suggested.